AricHasting/senior-software
# Added New Avatars and GUI control

### Goal
In our last meeting with Kuttal, she mentioned that she wanted to see options to use different avatars, including animated avatars. As a group, we also discussed that we would like to have a GUI element for Avatar control.
### Solution
**Multiple Avatars**
    I used a free online animation maker named 'animaker' to design two simple animated avatars, one male and one female. Each avatar model has the same set of states as the original avatar. The female model is set to default, and the user can switch between models using the command '/model <name>' where name is one of the valid avatar names. The avatar_util.py file now defines an enumerable type 'Avatar' that lists all valid avatar models:
`class Avatar(Enum):
    MALE = "male"
    FEMALE = "female"
    THE_BEST_AVATAR = "paul"`
**GUI Control**
I added two combo-boxes to the Avatar_widget class to control the avatar model and state. The combo-boxes are not visible when the program first starts running. I added the method sendHagrid() to the gui.py file which sets global variable Im_a_wizard_harry to true and makes the avatar combo-boxes visible. This method is now called whenever a user becomes the Wizard user by sending the '/wizard' command. Changing the value in a combo-box will cause the local model/state of the avatar to change, and will send a command message to the Dorothy user so that her avatar changes as well. The values used in the combo-boxes are determined by the corresponding enumerable types defined in the avatar_util.py file. Using text commands to change the avatar model/state still works, and will update the value of the corresponding combo-box to match.
###Verification
1. Start server
2. Start two gui.py instances
3. Enter names for each user
4. Use '/wizard' command to elevate one user to wizard
    * Two combo-boxes should appear under the avatar for the wizard user
5. As the Wizard, send "/avatar confused"
    * Avatar for both users should change to state confused, this was original functionality
6. As the Wizard, send "/model male"
    * Avatar for both users should change model to male
7. As the Wizard, use the bottom combo-box to select state 'happy'
    * Avatar for both users should change to state happy
8. As the Wizard, user the top combo-box to select model 'paul'
    * Avatar for both users should change model to my beautiful face.
